tests/unit/adapters/osz_archive_test.py

```
@@ -0,0 +1,308 @@
+    data = _archive({f"file{i}.txt": b"x" for i in range(5)})
+
+    with pytest.raises(OszArchiveError, match="members"):
+        read_osz(data, limits=limits)
+
+
+# --- difficulty counts ---------------------------------------------------
+
+
+def test_an_archive_with_no_difficulty_is_rejected() -> None:
+    """An archive carrying no ``.osu`` at all is not a beatmap."""
+    with pytest.raises(OszArchiveError, match="no .osu"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)